packages/admin/cms-admin/src/dam/DataGrid/fileUpload/useDamFileUpload.tsx

```diff
@@ -427,27 +423,29 @@ export const useDamFileUpload = (options: UploadDamFileOptions): FileUploadApi =
                     };
 
                     const uploadParams = {
-                        apiClient: createHttpClient(apiUrl),
+                        apiUrl,
                         data: uploadConfig,
-                        cancelToken: cancelUpload.current.token,
                         options: { onUploadProgress },
                         damBasePath,
                     };
 
-                    const response: { data: { id: string } } =
-                        duplicateAction === "replace" ? await replaceByFilenameAndFolder(uploadParams) : await upload(uploadParams);
+                    const uploadResult =
+                        duplicateAction === "replace"
+                            ? await replaceByFilenameAndFolder<{ id: string }>(uploadParams)
+                            : await upload<{ id: string }>(uploadParams);
+
+                    const response = await uploadResult.promise;
 
                     uploadedFiles.push({ id: response.data.id, parentId: targetFolderId, type: "file", file });
                 } catch (err) {
                     errorOccurred = true;
-                    const typedErr = err as AxiosError<{ error: string; message: string; statusCode: number } | string>;
+                    const typedErr = err as { response?: { data?: unknown }; request?: unknown };
```

Comment:
IMO we should remove the type cast here since we have type guards. Casting the type and then checking it via type guard is strange.